### ISSUE TITLE
0x80070057 error while executing table insert from PCL

### ISSUE THREAD

**Issue by rpedersen** [Azure/azure-storage-net#207]
UWP app is calling into PCL library that is referencing WindowsAzure.Storage 6.1.1-preview

Code compiles just fine but at runtime we get a odd message saying that the CreateIfNotExistsAsync method isn't found. 

```
private async Task<CloudTable> GetDriverCloudTableAsync()
{
    var driverCloudTable = _client.GetTableReference(DriverTableName);
   
    // adding the task delay in here so that we can keep the method as async with the line below it 
    // commented out so the method doesn't explode at runtime.
    await Task.Delay(1000); // this line is just fluff -- ignore

    //await driverCloudTable.CreateIfNotExistsAsync();

    return driverCloudTable;
}
```

Breaking just after the call to "GetTableReference" we can use the CloudTable instance inside the immediate window to attempt to execute the "CreateIfNotExistsAsync" method. When the attempt to execute the method is done this way we naturally get an unhandled exception with a Native View value of "Unable to evaluate the expression. Operation not supported. Unknown error: 0x80070057."

**Comment by pemari-msft** [comment]
Hi @rpedersen, thanks for the report. However, at this time, UWP is not supported by Azure Storage. There's a bug in the underlying UWP HTTP stack that often causes authentication failures in our scenarios. You can try using the universal DLL in version 6.1.0 which should give you better results. We're currently waiting for the UWP team to fix the HTTP bug and then we'll work to get UWP officially supported by Storage. Hope this helps!

**Comment by rpedersen** [comment]
Not sure that we are talking about the same things. When I put the Azure Storage calls inside of a Universal Windows Class Library it executes just fine. I am able to talk to table storage without issue, although, I am not executing at any scale. When I move the code out of the Universal Windows Class Library and into a Portable Class Library, because I need to reuse the code in a command line utility too, the code will compile but then at runtime I get a method not found exception because the "CreateIfNotExistsAsync" above doesn't seem to exist. This is all happening well before any HTTP call is executed.

**Comment by pemari-msft** [comment]
After some more offline discussion, we were not able to repro this. Please do let us know if you are able to get that repro!

**Comment by jpoon** [comment]
Repro: https://github.com/jpoon/AzureStoragePortableLibraryTest

Interfacing with Azure Storage within the UWP project works. Once we put this logic into a PCL, things blow up.

**Comment by jpoon** [comment]
As per @pemari-msft in an offline email:

> Because of how the dependency system works for PCLs, your App is only pulling in the WindowsAzure.Storage.dll from the Win8 target (the universal app target) and never the PCL target. It does this because even though you have a PCL, you’re only using it from the UWP context and so it thinks it can just swap out the PCL dll with the Universal dll (the aforementioned bait and switch) when the versions and namespaces are the same. This is verifiable using a tool such as listdlls: 

>C:\Users\pemari\Desktop>Listdlls.exe 4480
ListDLLs v3.1 - List loaded DLLs
0x00000000084e0000  0x8000    C:\Users\pemari\Desktop\test\AzureStoragePortableLibraryTest-master2\AzureStoragePortableLibraryTest-master\UWP_App\bin\x86\Debug\AppX\PortableLibrary.dll
0x000000000b240000  0xc6000   C:\Users\pemari\Desktop\test\AzureStoragePortableLibraryTest-master2\AzureStoragePortableLibraryTest-master\UWP_App\bin\x86\Debug\AppX\Microsoft.WindowsAzure.Storage.dll

> So now you just have the UWP dll in your app, which is why that button works. Then when you try to use the PCL buttons, it calls into the UWP dll (which works for account creation, which is why you don’t get the exn about how you’re using SharedKey in the PCL context) but when it looks for the `ExecuteAsync` signature, it doesn’t find it. So you get the exception you see. The reason for this method not existing is because the old universal (win8) library uses `IAsyncOperation` (for WinMD support) as a return type, whereas the new PCL uses `Tasks`.

tldr; 
1. remove explicit AzureStorage dependency in the UWP, 
2. no sharedkey support in the AzureStorage PCL so use SAS
